srcCpp/Infrastructure_micro.cxx

```diff
@@ -369,6 +369,71 @@ bool PerftestConfigureTransport(
         DDS_DomainParticipantQos &qos,
         ParameterManager *_PM)
 {
+    RTRegistry *registry = DDSDomainParticipantFactory::get_instance()->get_registry();
+    struct RTPS_CrcClass custom_crc16 = {-1, NULL, CustomCRC_crc16};
+    struct RTPS_CrcClass custom_crc32 = {-2, NULL, CustomCRC_crc32};
+    struct RTPS_CrcClass custom_crc64 = {-3, NULL, CustomCRC_crc64};
+    struct RTPS_CrcClass custom_crc128 = {-4, NULL, CustomCRC_crc128};
+
+    if (_PM->is_set("crc")) {
+        qos.protocol.compute_crc = RTI_TRUE;
+        qos.protocol.check_crc = RTI_TRUE;
+        qos.protocol.allowed_crc_mask = DDS_CRC_BUILTIN16 | DDS_CRC_BUILTIN32 | DDS_CRC_BUILTIN64;
+
+        // If we are goind to use custom CRC functions, we need to register them
+        if (_PM->is_set("customCrc")) {
+            qos.protocol.allowed_crc_mask |= DDS_CRC_CUSTOM16 | DDS_CRC_CUSTOM32 | DDS_CRC_CUSTOM64;
+
+            if (!registry->unregister(NETIO_DEFAULT_RTPS_NAME, NULL, NULL)) {
+                fprintf(stderr, "Failed to unregister rtps component\n");
+                return false;
+            }
+
+            OSAPI_Heap_allocate_struct(&transport.transportConfig.rtps_property, struct RTPS_InterfaceFactoryProperty);
```

Comment:
Add a comment here stating that the struct is freed on the TransportConfig destructor

Reply:
Done